ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Ledger.hs

```diff
@@ -519,71 +522,242 @@ data HardForkLedgerWarning xs =
     -- /before/ this lower bound, the node is misconfigured and will likely
     -- not work correctly. This should be taken care of as soon as possible
     -- (before the transition happens).
-  | HardForkLedgerWarningUnexpectedTransition EraParams EpochNo
+  | HardForkWarningTransitionMismatch (EraIndex xs) EraParams EpochNo
+
+    -- | Transition in the final era
+    --
+    -- The final era should never confirm any transitions
+  | HardForkWarningTransitionInFinalEra (EraIndex xs) EpochNo
```

Comment:
This `(EraIndex xs)` is then the final era?

Reply:
Yes. Added "For clarity, we also record the index of that final era." to the docstring.